packages/hash/api/src/graphql/resolvers/ontology/entity-type.ts

```diff
@@ -0,0 +1,122 @@
+import { ApolloError } from "apollo-server-express";
+import { AxiosError } from "axios";
+
+import {
+  PersistedEntityType,
+  MutationCreateEntityTypeArgs,
+  MutationUpdateEntityTypeArgs,
+  QueryGetEntityTypeArgs,
+  Resolver,
+} from "../../apiTypes.gen";
+import { GraphQLContext } from "../../context";
+import { EntityTypeModel } from "../../../model";
+import { nilUuid } from "../../../model/util";
+import { entityTypeModelToGQL } from "./model-mapping";
+
+export const createEntityType: Resolver<
+  Promise<PersistedEntityType>,
+  {},
+  GraphQLContext,
+  MutationCreateEntityTypeArgs
+> = async (_, params, { dataSources }) => {
+  const { graphApi } = dataSources;
+  const { accountId, entityType } = params;
+
+  const createdEntityTypeModel = await EntityTypeModel.create(graphApi, {
+    accountId,
+    schema: entityType,
+  }).catch((err: AxiosError) => {
+    if (err.response?.status === 409) {
+      throw new ApolloError(
+        `Entity type with the same URI already exists. [URI=${entityType.$id}]`,
+        "CREATION_ERROR",
+      );
+    }
+
+    throw new ApolloError(`Couldn't create entity type.`, "CREATION_ERROR");
+  });
+
+  return entityTypeModelToGQL(createdEntityTypeModel);
+};
+
+export const getAllLatestEntityTypes: Resolver<
+  Promise<PersistedEntityType[]>,
+  {},
+  GraphQLContext,
+  {}
+> = async (_, __, { dataSources }) => {
+  const { graphApi } = dataSources;
+
+  const allLatestEntityTypeModels = await EntityTypeModel.getAllLatest(
+    graphApi,
+    {
+      /** @todo Replace with User from the request */
+      accountId: nilUuid,
+    },
+  ).catch((err: AxiosError) => {
+    throw new ApolloError(
+      `Unable to retrieve all latest entity types. ${err.response?.data}`,
+      "GET_ALL_ERROR",
+    );
+  });
+
+  return allLatestEntityTypeModels.map((entityTypeModel) =>
+    entityTypeModelToGQL(entityTypeModel),
+  );
```

Comment:
Nit:
```suggestion
  return allLatestEntityTypeModels.map(entityTypeModelToGQL);
```

Reply:
Addressed in ad87f89d3108c79586332e1f59d5f35cef31cbd5 for all ontology types